File: 5xHIT/Decrypt/decrypt.py
"""
Author: Markus Reichl
Version: 30.11.2016
"""

defa(b,c="advent",d=""):fore,finenumerate(b):d+=chr((ord(f)+ord(c[e%len(c)]))%int(0b1111110))returnd\n<UmceO\]Kc\h\jpOK

# START SOLVED
def decrypt(message, key):
    __crypt = ""

    for i, char in enumerate(message):
        key_char = ord(key[i % len(key)])
        __crypt += chr((ord(char) - key_char) % 126)

    return __crypt


if __name__ == "__main__":
    decrypted = "You solved my puzzle!"

    ec = a(decrypted, "advent")
    print(ec)

    dc = decrypt(encrypted, "advent")
    print(dc)
# END SOLVED

File: 3xHIT/Aufgaben/Decrypt/src/Decrypt.java
/**
 * @author Markus Reichl
 * @version 30.11.2016
 *
 * String a(String d){String c="BCDEFGHIJKLMNOPQRSTUVWXYZA";String b="ABCDEFGHIJKLMNOPQRSTUVWXYZ";String e="";d=d.toUpperCase();for (int i = 0; i < d.length(); i++){char f=d.charAt(i);if (b.indexOf(f) > -1) {e+=c.charAt(b.indexOf(f));}else{e+=f;}}return e;} \n "ZPV TPMWFE NZ QVAAMF"
 */


class Decrypt {
    private String alphabet;
    private String key;

    public static void main(String[] args) {
        String alphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
        String key = "BCDEFGHIJKLMNOPQRSTUVWXYZA";

        System.out.println(encrypt(alphabet, key, "You solved my puzzle"));
        System.out.println(encrypt(key, alphabet, "ZPV TPMWFE NZ QVAAMF"));
    }

    private static String encrypt(String alphabet, String key, String message) {
        String encrypted = "";

        message = message.toUpperCase();

        for (int i = 0; i < message.length(); i++){
            char c = message.charAt(i);

            if (alphabet.indexOf(c) > -1) {
                encrypted += key.charAt(alphabet.indexOf(c));
            } else {
                encrypted += c;
            }
        }

        return encrypted;
    }
}
